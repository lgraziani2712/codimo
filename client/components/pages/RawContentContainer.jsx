/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import styled from 'styled-components';

const RawContentContainer = styled.div`
  margin: 0 auto;
  width: 980px;
`;

export default RawContentContainer;
