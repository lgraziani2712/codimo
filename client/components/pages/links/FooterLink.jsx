/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

const Link = styled.a`
  color: snow;
  font-weight: bold;
  text-decoration: none;
  &:hover {
    color: #dadada;
    text-decoration: underline;
  }
`;

type Props = {|
  href: string,
  text: string,
|};
const FooterLink = ({ href, text }: Props) => (
  <Link href={href} target="_blank">{text}</Link>
);

export default FooterLink;
