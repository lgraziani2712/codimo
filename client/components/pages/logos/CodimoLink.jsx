/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled, { keyframes } from 'styled-components';
import { Link } from 'react-router-dom';

import LogoTitleURL from './logo-title.png';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;
const HomeLink = styled(Link)`
  align-items: center;
  display: flex;
  text-decoration: none;
  &:active img {
    animation: ${rotate360} 0.245s linear infinite;
  }
`;
const Title = styled.span`
  color: #f9524e;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 3px;
  text-shadow: 0 1px 2px rgba(85, 95, 110, 0.4);
`;
const ShadowedImg = styled.img`
  filter: drop-shadow(0 1px 2px rgba(85, 95, 110, 0.4));
`;

const CodimoLink = () => (
  <HomeLink to="/">
    <ShadowedImg src={LogoTitleURL} />
    <Title>{'CÃ³dimo'}</Title>
  </HomeLink>
);

export default CodimoLink;
