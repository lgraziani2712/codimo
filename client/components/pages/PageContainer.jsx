/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled, { css } from 'styled-components';

import { COLOR_PALETTE } from 'core/constants/colors';

import FooterLink from './links/FooterLink';
import RawContentContainer from './RawContentContainer';

const Wrapper = styled.div`
  margin-bottom: 10rem;
  user-select: initial;
`;
const ContentContainer = styled(RawContentContainer)`
  display: flex;
  justify-content: space-between;
`;
const fixedBottomMixin = css`
  bottom: 0;
  height: 10rem;
  margin: 0;
  position: fixed;
  width: 100%;
  z-index: -1;
`;
const Footer = styled.div`
  align-items: center;
  background: ${COLOR_PALETTE.blue.normal};
  color: white;
  display: flex;
  ${fixedBottomMixin}
`;

type Props = {|
  children?: React.Element<*>,
|};
const PageContainer = ({ children }: Props) => (
  <Wrapper>
    <div>
      {children}
    </div>
    <Footer>
      <ContentContainer>
        <span>
          {'Equipo de desarrollo de Códimo. '}
          <FooterLink href="https://www.unrn.edu.ar/" text="UNRN" />
          {'.'}
        </span>
        <span>
          {'Distribuído bajo licencia '}
          <FooterLink
            href="https://github.com/lgraziani2712/codimo/blob/master/LICENSE"
            text="MIT"
          />
          {'.'}
        </span>
      </ContentContainer>
    </Footer>
  </Wrapper>
);

export default PageContainer;
