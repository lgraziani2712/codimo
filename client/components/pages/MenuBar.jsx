/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import { type Codimo$Route } from 'core/ui/CodimoRouter';

import CodimoLink from './logos/CodimoLink';
import GithubLink from './logos/GithubLink';
import HeaderLink from './links/HeaderLink';
import RawContentContainer from './RawContentContainer';

const Background = styled.div`
  background: rgba(240, 240, 240, 0.9);
  box-shadow: 0 0 4px 4px rgba(85, 95, 110, 0.4);
  height: 55px;
  position: relative;
  width: 100%;
  z-index: 1;
`;
const ContentContainer = styled(RawContentContainer)`
  display: flex;
  height: 100%;
  justify-content: space-between;
`;
const LinksContainer = styled.div`
  display: flex;
  flex-grow: 1;
  justify-content: space-between;
`;
const LocalLinksContainer = styled.ul`
  display: flex;
  margin: 0;
  padding: 0 0 0 1.5rem;
`;
const ExternalLinksContainer = styled.div`
  align-items: center;
  display: flex;
  flex-grow: 1;
  justify-content: flex-end;

  & > * {
    padding-left: 0.5rem;
  }
`;

type Props = {|
  routes: Array<Codimo$Route>,
|};
/**
 * CÃ³dimo client menu bar.
 *
 * @version 1.1.0
 * @param {Array<Codimo$Route>} routes List of routes.
 * @return {React$Element}             Menu bar element.
 */
const MenuBar = ({ routes }: Props) => (
  <Background>
    <ContentContainer>
      <CodimoLink />

      <LinksContainer>
        <LocalLinksContainer>
          {routes.map((route) => {
            const path =
              !route.children
                ? `/${route.activityName}`
                // $FlowDoNotDisturb difficulty prop exist if the route has children
                : `/${route.activityName}/${route.difficulty}/${route.children[0].path}`;

            return (
              <HeaderLink
                key={path}
                to={path}
                title={route.title}
              />
            );
          })}
        </LocalLinksContainer>

        <ExternalLinksContainer>
          <GithubLink />
        </ExternalLinksContainer>
      </LinksContainer>
    </ContentContainer>
  </Background>
);

export default MenuBar;
