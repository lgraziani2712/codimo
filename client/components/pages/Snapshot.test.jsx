/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 * @jest-environment node
 */
import React from 'react';
import {
  MemoryRouter as Router,
} from 'react-router-dom';
import renderer from 'react-test-renderer';

import 'test/BlocklyMock';
import 'jest-styled-components';

import CodimoLink from './logos/CodimoLink';
import FooterLink from './links/FooterLink';
import GithubLink from './logos/GithubLink';
import RawContentContainer from './RawContentContainer';
import MenuBar from './MenuBar';
import PageContainer from './PageContainer';

jest.mock('core/helpers/randomizers');

describe('Components snapshots', () => {
  it('renders the links correctly', () => {
    const footerLinkTree = renderer.create(
      <FooterLink href="http://helloworld.test" text="Hello dear" />,
    ).toJSON();
    const codimoLinkTree = renderer.create(<Router><CodimoLink /></Router>).toJSON();
    const githubLinkTree = renderer.create(<GithubLink />).toJSON();

    expect(footerLinkTree).toMatchSnapshot();
    expect(codimoLinkTree).toMatchSnapshot();
    expect(githubLinkTree).toMatchSnapshot();
  });
  it('renders the raw components correctly', () => {
    const rawContentContainerTree = renderer.create(<RawContentContainer />).toJSON();

    expect(rawContentContainerTree).toMatchSnapshot();
  });
  it('renders the MenuBar correctly', () => {
    const tree = renderer.create(<Router><MenuBar routes={[]} /></Router>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders the PageContainer correctly', () => {
    const tree = renderer.create(<PageContainer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
