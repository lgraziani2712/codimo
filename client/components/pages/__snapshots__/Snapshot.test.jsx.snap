// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshots renders the MenuBar correctly 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-decoration: none;
}

.c3:active img {
  -webkit-animation: iVXCSc 0.245s linear infinite;
  animation: iVXCSc 0.245s linear infinite;
}

.c5 {
  color: #f9524e;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 3px;
  text-shadow: 0 1px 2px rgba(85,95,110,0.4);
}

.c4 {
  filter: drop-shadow(0 1px 2px rgba(85,95,110,0.4));
}

.c9 {
  height: 22px;
  position: relative;
  vertical-align: middle;
}

.c9 path {
  fill: rgba(0,0,0,0.6);
}

a:hover > .c9 path {
  fill: rgba(0,0,0,0.7);
}

.c2 {
  margin: 0 auto;
  width: 980px;
}

.c0 {
  background: rgba(240,240,240,0.9);
  box-shadow: 0 0 4px 4px rgba(85,95,110,0.4);
  height: 55px;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0 0 0 1.5rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c8 > * {
  padding-left: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <a
      className="c3"
      href="/"
      onClick={[Function]}
    >
      <img
        className="c4"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <span
        className="c5"
      >
        Códimo
      </span>
    </a>
    <div
      className="c6"
    >
      <ul
        className="c7"
      />
      <div
        className="c8"
      >
        <a
          href="https://github.com/lgraziani2712/codimo"
          target="_blank"
        >
          <svg
            aria-labelledby="github-svg"
            className="c9"
            role="img"
            viewBox="0 0 32.6 31.8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="github-svg"
            >
              Github
            </title>
            <path
              d="M16.3 0C7.3 0 0 7.3 0 16.3c0 7.2 4.7 13.3 11.1 15.5.8.1 1.1-.4 1.1-.8v-2.8c-4.5 1-5.5-2.2-5.5-2.2-.7-1.9-1.8-2.4-1.8-2.4-1.5-1 .1-1 .1-1 1.6.1 2.5 1.7 2.5 1.7 1.5 2.5 3.8 1.8 4.7 1.4.1-1.1.6-1.8 1-2.2-3.6-.4-7.4-1.8-7.4-8.1 0-1.8.6-3.2 1.7-4.4-.1-.3-.7-2 .2-4.2 0 0 1.4-.4 4.5 1.7 1.3-.4 2.7-.5 4.1-.5 1.4 0 2.8.2 4.1.5 3.1-2.1 4.5-1.7 4.5-1.7.9 2.2.3 3.9.2 4.3 1 1.1 1.7 2.6 1.7 4.4 0 6.3-3.8 7.6-7.4 8 .6.5 1.1 1.5 1.1 3V31c0 .4.3.9 1.1.8 6.5-2.2 11.1-8.3 11.1-15.5C32.6 7.3 25.3 0 16.3 0z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components snapshots renders the PageContainer correctly 1`] = `
.c4 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c4:hover {
  color: #dadada;
  text-decoration: underline;
}

.c3 {
  margin: 0 auto;
  width: 980px;
}

.c0 {
  margin-bottom: 10rem;
  -webkit-user-select: initial;
  -moz-user-select: initial;
  -ms-user-select: initial;
  user-select: initial;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #428bca;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  bottom: 0;
  height: 10rem;
  margin: 0;
  position: fixed;
  width: 100%;
  z-index: -1;
}

<div
  className="c0"
>
  <div />
  <div
    className="c1"
  >
    <div
      className="c2 c3"
    >
      <span>
        Equipo de desarrollo de Códimo. 
        <a
          className="c4"
          href="https://www.unrn.edu.ar/"
          target="_blank"
        >
          UNRN
        </a>
        .
      </span>
      <span>
        Distribuído bajo licencia 
        <a
          className="c4"
          href="https://github.com/lgraziani2712/codimo/blob/master/LICENSE"
          target="_blank"
        >
          MIT
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;

exports[`Components snapshots renders the links correctly 1`] = `
.c0 {
  color: snow;
  font-weight: bold;
  text-decoration: none;
}

.c0:hover {
  color: #dadada;
  text-decoration: underline;
}

<a
  className="c0"
  href="http://helloworld.test"
  target="_blank"
>
  Hello dear
</a>
`;

exports[`Components snapshots renders the links correctly 2`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-decoration: none;
}

.c0:active img {
  -webkit-animation: iVXCSc 0.245s linear infinite;
  animation: iVXCSc 0.245s linear infinite;
}

.c2 {
  color: #f9524e;
  font-size: 1.5rem;
  font-weight: bold;
  margin-left: 3px;
  text-shadow: 0 1px 2px rgba(85,95,110,0.4);
}

.c1 {
  filter: drop-shadow(0 1px 2px rgba(85,95,110,0.4));
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
>
  <img
    className="c1"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <span
    className="c2"
  >
    Códimo
  </span>
</a>
`;

exports[`Components snapshots renders the links correctly 3`] = `
.c0 {
  height: 22px;
  position: relative;
  vertical-align: middle;
}

.c0 path {
  fill: rgba(0,0,0,0.6);
}

a:hover > .c0 path {
  fill: rgba(0,0,0,0.7);
}

<a
  href="https://github.com/lgraziani2712/codimo"
  target="_blank"
>
  <svg
    aria-labelledby="github-svg"
    className="c0"
    role="img"
    viewBox="0 0 32.6 31.8"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="github-svg"
    >
      Github
    </title>
    <path
      d="M16.3 0C7.3 0 0 7.3 0 16.3c0 7.2 4.7 13.3 11.1 15.5.8.1 1.1-.4 1.1-.8v-2.8c-4.5 1-5.5-2.2-5.5-2.2-.7-1.9-1.8-2.4-1.8-2.4-1.5-1 .1-1 .1-1 1.6.1 2.5 1.7 2.5 1.7 1.5 2.5 3.8 1.8 4.7 1.4.1-1.1.6-1.8 1-2.2-3.6-.4-7.4-1.8-7.4-8.1 0-1.8.6-3.2 1.7-4.4-.1-.3-.7-2 .2-4.2 0 0 1.4-.4 4.5 1.7 1.3-.4 2.7-.5 4.1-.5 1.4 0 2.8.2 4.1.5 3.1-2.1 4.5-1.7 4.5-1.7.9 2.2.3 3.9.2 4.3 1 1.1 1.7 2.6 1.7 4.4 0 6.3-3.8 7.6-7.4 8 .6.5 1.1 1.5 1.1 3V31c0 .4.3.9 1.1.8 6.5-2.2 11.1-8.3 11.1-15.5C32.6 7.3 25.3 0 16.3 0z"
    />
  </svg>
</a>
`;

exports[`Components snapshots renders the raw components correctly 1`] = `
.c0 {
  margin: 0 auto;
  width: 980px;
}

<div
  className="c0"
/>
`;
